torchvision/datasets/celeba.py

```diff
@@ -0,0 +1,171 @@
+        ("0B7EVK8r0v71pblRyaVFSWGxPY0U", "75e246fa4810816ffd6ee81facbd244c", "list_attr_celeba.txt"),
+        ("1_ee_0u7vcNLOfNLegJRHmolfH5ICW-XS", "32bd1bd63d3c78cd57e08160ec5ed1e2", "identity_CelebA.txt"),
+        ("0B7EVK8r0v71pbThiMVRxWXZ4dU0", "00566efa6fedff7a56946cd1c10f1c16", "list_bbox_celeba.txt"),
+        ("0B7EVK8r0v71pd0FJY3Blby1HUTQ", "cc24ecafdb5b50baae59b03474781f8c", "list_landmarks_align_celeba.txt"),
+        # ("0B7EVK8r0v71pTzJIdlJWdHczRlU", "063ee6ddb681f96bc9ca28c6febb9d1a", "list_landmarks_celeba.txt"),
+        ("0B7EVK8r0v71pY0NSMzRuSXJEVkk", "d32c9cbf5e040fd4025c592c306e6668", "list_eval_partition.txt"),
+    ]
+
+    def __init__(self, root,
+                 split="train",
+                 target_type="attr",
+                 transform=None, target_transform=None,
+                 download=False):
+        import pandas
+        self.root = os.path.expanduser(root)
+        self.split = split
+        if isinstance(target_type, list):
+            self.target_type = target_type
+        else:
+            self.target_type = [target_type]
+        self.transform = transform
+        self.target_transform = target_transform
+
+        if download:
+            self.download()
+
+        if not self._check_integrity():
+            raise RuntimeError('Dataset not found or corrupted.' +
+                               ' You can use download=True to download it')
+
+        self.transform = transform
+        self.target_transform = target_transform
+
+        if split.lower() == "train":
+            split = 0
+        elif split.lower() == "valid":
+            split = 1
+        elif split.lower() == "test":
+            split = 2
+        else:
+            raise ValueError('Wrong split entered! Please use split="train" '
+                             'or split="valid" or split="test"')
+
+        with open(os.path.join(self.root, self.base_folder, "list_eval_partition.txt"), "r") as f:
+            splits = pandas.read_csv(f, delim_whitespace=True, header=None, index_col=0)
+
+        with open(os.path.join(self.root, self.base_folder, "identity_CelebA.txt"), "r") as f:
+            self.identity = pandas.read_csv(f, delim_whitespace=True, header=None, index_col=0)
+
+        with open(os.path.join(self.root, self.base_folder, "list_bbox_celeba.txt"), "r") as f:
+            self.bbox = pandas.read_csv(f, delim_whitespace=True, header=1, index_col=0)
+
+        with open(os.path.join(self.root, self.base_folder, "list_landmarks_align_celeba.txt"), "r") as f:
+            self.landmarks_align = pandas.read_csv(f, delim_whitespace=True, header=1)
+
+        with open(os.path.join(self.root, self.base_folder, "list_attr_celeba.txt"), "r") as f:
+            self.attr = pandas.read_csv(f, delim_whitespace=True, header=1)
+
+        mask = (splits[1] == split)
+        self.filename = splits[mask].index.values
+        self.identity = torch.LongTensor(self.identity[mask].values)
```

Comment:
`torch.LongTensor` is the legacy constructor, could you use instead `torch.as_tensor`? If you want to force it to be `int64`, you can do `torch.as_tensor(..., dtype=torch.int64)`.